test/lib/code_corps/helpers/policy_test.exs

```diff
@@ -0,0 +1,221 @@
+      organization = insert(:organization)
+      changeset = %Changeset{changes: %{organization_id: organization.id}}
+      result = Policy.get_organization(changeset)
+      assert result.id == organization.id
+      assert result.name == organization.name
+    end
+
+    test "return nil for structs with no organization_id" do
+      assert Policy.get_organization(%{foo: "bar"}) == nil
+    end
+
+    test "return nil for any" do
+      assert Policy.get_organization("foo") == nil
+    end
+  end
+
+
+  describe "get_project/1" do
+    test "return project if the project_id is defined on the struct" do
+      project = insert(:project)
+      project_category = insert(:project_category, project: project)
+      result = Policy.get_project(project_category)
+      assert result.id == project.id
+      assert result.title == project.title
+    end
+
+    test "return project if the project_id is defined on the changeset" do
+      project = insert(:project)
+      changeset = %Changeset{changes: %{project_id: project.id}}
+      result = Policy.get_project(changeset)
+      assert result.id == project.id
+      assert result.title == project.title
+    end
+
+    test "return nil for structs with no project_id" do
+      assert Policy.get_project(%{foo: "bar"}) == nil
+    end
+
+    test "return nil for any" do
+      assert Policy.get_project("foo") == nil
+    end
+  end
+
+  describe "get_role/1" do
+    test "should return a project user's role if it's defined" do
+      assert Policy.get_role(%ProjectUser{role: "admin"}) == "admin"
+    end
+    test "should return a changeset's role if it's defined" do
+      assert Policy.get_role(%Changeset{data: %{role: "contributor"}, types: %{role: :string}}) == "contributor"
+    end
+
+    test "should return nil if no role is defined on a project user" do
+      assert Policy.get_role(%ProjectUser{}) == nil
+    end
+
+    test "should return nil if no role is defined on a changeset" do
+      assert Policy.get_role(%Changeset{data: %{role: nil}, types: %{role: :string}}) == nil
+    end
+
+    test "should return nil if nil is passed in" do
+      assert Policy.get_role(nil) == nil
+    end
+  end
+
+  describe "get_task/1" do
+    test "should return task of a TaskSkill" do
+      task = insert(:task)
+      task_skill = insert(:task_skill, task: task)
+      result = Policy.get_task(task_skill)
+      assert result.id == task.id
+    end
+
+    test "should return task of a UserTask" do
+      task = insert(:task)
+      user_task = insert(:user_task, task: task)
+      result = Policy.get_task(user_task)
+      assert result.id == task.id
+    end
+
+    test "should return task of a Changeset" do
+      task = insert(:task)
+      changeset = %Changeset{changes: %{task_id: task.id}}
+      result = Policy.get_task(changeset)
+      assert result.id == task.id
+    end
+  end
+
+  describe "task_authored_by?/1" do
+    test "should return true if the Task user_id and User id match" do
```

Comment:
Again, probably word these as 

`"returns true if the user is the author of the task"`
`"returns false if the user is not the author of the task"`